ksql-serde/src/test/java/io/confluent/ksql/serde/delimited/KsqlDelimitedDeserializerTest.java

```diff
@@ -185,6 +190,26 @@ public void shouldDeserializedTopLevelPrimitiveTypeIfSchemaHasOnlySingleField()
     assertThat(result.get("id"), CoreMatchers.is(10));
   }
 
+  @Test
+  public void shouldDeserializeDecimal() {
+    // Given:
+    final PersistenceSchema schema = persistenceSchema(
+        SchemaBuilder.struct()
+          .field("cost", DecimalUtil.builder(4, 2))
+          .build()
+    );
+    final KsqlDelimitedDeserializer deserializer =
+        new KsqlDelimitedDeserializer(schema, recordLogger);
+
+    final byte[] bytes = "01.12".getBytes(StandardCharsets.UTF_8);
```

Comment:
We'd also correctly deserialize if the leading 0 were missing, right? Perhaps worth adding a test to ensure that.